monitoring/bentoml-plugins-arize/src/bentoml_plugins/arize/__init__.py

```diff
@@ -376,39 +353,37 @@ def export_schema(self):
 
         if self.model_version is None and self.model_id is None:
 
-            self.model_id = component_context.bento_name
+            self.model_id = f"{component_context.bento_name}:{self.name}"
             self.model_version = component_context.bento_version
 
         if self.environment is None:
             self.environment = Environments.PRODUCTION
 
         self._init_client()
 
-    def export_data(self):
+    def export_data(self, datas: dict[str, collections.deque[DataType]]) -> None:
         """
         Export data. This method should be called after all data is logged.
         """
-        assert (
-            len(set(len(q) for q in self._columns.values())) == 1
-        ), "All columns must have the same length"
         assert self.model_id is not None
         assert self.model_type is not None
         assert self.environment is not None
+
+        timestamp = datetime.datetime.now().timestamp()
+        prediction_id = trace_context.request_id
+        assert isinstance(prediction_id, int)
         while True:
             try:
-                record = {k: v.popleft() for k, v in self._columns.items()}
+                record = {k: v.popleft() for k, v in datas.items()}
```

Comment:
`break` based on catching an `Error` is probably not efficient. Raising error is typically expensive because it has to fill the stack trace. Do we have a breaking condition without error?

Reply:
Typically this would be faster. The length checking would be O(n), and the error catching is O(1).
But yeah, if every value here only has one element, it could be slower.